core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessageDuplicator.java

```diff
@@ -83,10 +85,19 @@ protected AbstractStreamMessageDuplicator(U publisher) {
      * this factory with.
      */
     public U duplicateStream() {
+        return duplicateStream(false);
+    }
+
+    /**
+     * Creates a new {@link U} instance that publishes data from the {@code publisher} you create
+     * this factory with. If you specify the {@code lastStream} as {@code true}, it will prevent further
+     * creation of duplicate stream.
+     */
+    public U duplicateStream(boolean lastStream) {
```

Comment:
How about adding a method like `freeze()` instead of `duplicateStream(boolean lastStream)`? I'm just thinking about the case that a user wants to finish a duplication without a subscriber.